lib/srv/desktop/tdp/proto.go

```diff
@@ -844,6 +847,10 @@ func decodeFileSystemObject(in peekReader) (FileSystemObject, error) {
 	if err != nil {
 		return FileSystemObject{}, trace.Wrap(err)
 	}
+	err = binary.Read(in, binary.BigEndian, &isEmpty)
+	if err != nil {
+		return FileSystemObject{}, trace.Wrap(err)
+	}
```

Comment:
You could just call `in.ReadByte()` here. Byte order doesn't matter if there's only one byte.